src/dpp/sslclient.cpp

```diff
@@ -237,11 +237,36 @@ ssl_client::ssl_client(const std::string &_hostname, const std::string &_port, b
 	keepalive(reuse)
 {
 #ifndef WIN32
-	signal(SIGALRM, SIG_IGN);
-	signal(SIGHUP, SIG_IGN);
-	signal(SIGPIPE, SIG_IGN);
-	signal(SIGCHLD, SIG_IGN);
-	signal(SIGXFSZ, SIG_IGN);
+	struct sigaction sa;
+	struct sigaction action;
+	memset(&action, 0, sizeof(action));
+	action.sa_handler = SIG_IGN;
+	action.sa_flags = SA_RESTART;
+
+	sigaction(SIGALRM, nullptr, &sa);
+	if (sa.sa_flags == 0 && sa.sa_handler == nullptr) {
```

Comment:
DRY, can you please make this a function and call the function with each signal name instead?

Reply:
Done